docs/Coiote_IoT_DM/firmware_update.md

```diff
@@ -0,0 +1,188 @@
+1. Choose between **Pull** and **Push**
+
+    **Pull method** (recommended): The LwM2M Client receives the URI of the file that is to be downloaded and pulls the file from it. 
+    
+    **Push method**: The LwM2M Server pushes the firmware file to the device.
+
+
+    !!! Info
+        **Pull** supports any of the following **transport types**:
+
+        - `CoAP` or `CoAPs` over `UDP`
+        - `CoAP` or `CoAPs` over `TCP`
+        - `HTTP` or `HTTPs`
+
+        **Push** uses `CoAPs` over `UDP` by default.
+
+    !!! Note
+        **Which transport protocol to choose?**
+        
+        Downloads using `CoAP(s)` over `UDP` tend to be slow due to limitation of the maximum CoAP Block size of 1024 bytes and the required acknowledgements for each Block transfer.
+
+        Choosing `CoAP(s)` over `TCP` or `HTTP(s)` usually results in faster download speeds.
+
+
+    ![Basic Firmware Update](images/Pull.png)
+
+
+1. Click **Schedule Update** from the bottom-right corner to trigger the Firmware Update process.
+
+    ![Schedule Update](images/schedule.png)
+
+
+## Download & Upgrade Process
+
+If the Firmware Update is scheduled successfully, the device starts **downloading** the firmware at the next practical opportunity. The actual firmware **update** starts once the integrity and authenticity of the firmware image has been validated by the LwM2M Client.
+
+![Upgrading](images/update-in-progress.png)
+
+The status of the firmware update is shown in the Resources: **State** `/5/*/3` & **Update Result** `/5/*/5`. If no errors arise, the update process follows the following pattern:
+
+1. **Downloading** `state 1` & `update result 0` 
+2. **Downloaded** `state 2` & `update result 0` 
+3. **Updating** `state 3` & `update result 0` 
+4. **Updated** `state 0` & `update result 1` 
+
+
+!!! info
+    When the device is updating its firmware, it will deregister and reboot using the new firmware. This process may time multiple minutes.
+
+    ![Deregistered while upgrading](images/deregistered.png)
+
+
+If the update was successfully performed, the **State** `/5/*/3` returns to `0` and the **Update Results** `/5/*/5` reports `1`.
+
+![Firmware upgraded](images/result1.png)
+
+The **Update list** also reports: `Success`.
+
+![Successful update](images/success.png)
+
+
+
+## Troubleshooting
+
+* ### Insufficient flash
+
+    A common update error is due to insufficient flash memory, indicated by **Update Result** `2`.
+
+    The Anjay build file can be quite large in size, particularly when using the `.hex` file (`app_signed.hex`). For updating the firmware, it’s better to use the `app_update.bin` file which is about half the size of the `.hex` file.
+
+    !!! Warning
+        The **Nordic Thingy:91** is very restricted in its flash size due to the Nordic Bootloader. The maximum firmware size is `421kB`.
+
+* ### Firmware Update only works over CoAP, not over CoAPs
+
+    !!! Warning
+        Possible issue when working with the **Zephyr LwM2M Client**.
```

Comment:
This might actually be a common issue on any device, so it might be worth describing it in a generic manner before the detailed instructions for the Zephyr client.

I'd propose something like:

> For the CoAPs transfer to work, the LwM2M Client shall use the same security credentials (i.e., PSK or certificates) as those used for the management interface. This is the default behavior of the Anjay client (as long as hostnames in the URI match), but you might need to configure it explicitly when using other implementations.
>
> [info box] Below is a configuration instruction for the Zephyr LwM2M Client.